build/rewrite.go

```diff
@@ -460,12 +467,21 @@ func sortStringLists(f *File, info *RewriteInfo) {
 					continue
 				}
 				context := rule + "." + key.Name
-				if !tables.IsSortableListArg[key.Name] || tables.SortableBlacklist[context] || !f.Build {
+
+				// Drop out if sorting is disabled, if we don't have a build file, or if
+				// this context is in the blacklist.
+				if disabled("unsafesort") || !f.Build || tables.SortableBlacklist[context] {
 					continue
 				}
-				if disabled("unsafesort") && !tables.SortableWhitelist[context] && !allowedSort(context) {
+
+				// Drop out unless this field name is whitelisted, we're explicitly
+				// allowed to sort it for debugging reasons, or its field name is
+				// explicitly allowed.
+				if !tables.SortableWhitelist[context] && !allowedSort(context) && !tables.IsSortableListArg[key.Name] {
```

Comment:
For clarification: is there any behavior change here?

Reply:
There are some minor behavior changes here:

- If "unsafesort" is disabled, then this makes this rewrite _never_ apply, whereas at head it will still apply if allowedSort returns true or the field is in SortableWhitelist, which seemed inconsistent with the semantics implied by the "disabled" method. Note that every other case here already drops out immediately if unsafesort is disabled.

- This makes SortableWhitelist and allowedSort allow a field to be sorted even if the given key.Name is not in IsSortableListArg: the current behavior seems to violate the implicit semantics of an explicit SortableWhitelist.